streams/spot_meta.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/python3
+""" Spotify Metadata Translator
+    This file reads/translates the metadata coming from vollibrespot and
+    writes the useful information to currentSong
+"""
+import socket
+import json
+import re
+import argparse
+
+parser = argparse.ArgumentParser(description="spotify metadata interpreter")
+parser.add_argument('metaport', help='port to read from (see metadata-port in config.toml)')
+parser.add_argument('cs_loc', help='Folder to write/create the currentSong file')
+parser.add_argument('--verbose', '-v', action='count', default=None, help='Prints full metadata')
+args = parser.parse_args()
+
+info = {
+'state': 'stopped',
+'artist': '',
+'album': '',
+'track': '',
+'img_url': ''
+}
+
+# Write to currentSong so you clear previous metadata from i.e. shairport or old Spotify instances
+with open(f'{args.cs_loc}/currentSong', 'w') as csi:
+  csi.write(str(info))
+
+metasocket = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+metasocket.bind(('', int(args.metaport)))
+prev_state = ''
+while True: # Need to check how to kill the script, look at processing load, etc. NEED TO VERIFY
+  message, address = metasocket.recvfrom(1024)
+  if message:
+    decoded = message.decode('utf-8', 'replace')
+    parsed = re.split('\n|\r', decoded)
+    if 'kSp' not in decoded:
+      data = {}
+      for item in parsed:
+        try:
+          data = json.loads(item)
+        except:
+          pass
+        if 'state' in data:
+          simplestate = str(data['state'])
+          ss = simplestate[11:].strip("'}")
+          if 'play' in ss:
+            info['state'] = 'playing' # JavaScript looks for 'playing', not 'play'
+          else:
+            info['state'] = ss
+        elif 'metadata' in data:
+          info['artist'] = data['metadata'].get('artist_name') # .get defaults to 'None'
+          info['album'] = data['metadata'].get('album_name') # if nothing found
+          info['track'] = data['metadata'].get('track_name')
+          al = data['metadata'].get('albumartId')
+          if al[0] is not None:
+            info['img_url'] = 'https://i.scdn.co/image/' + al[0]
+          else:
+            info['img_url'] = None
+    if args.verbose:
+      print(decoded)
+    elif info['state'] != prev_state:
+      print(f"{info['state']}: {info['track']}")
+      prev_state = info['state']
```

Comment:
Shouldn't this print out no matter what? I think we only get a message every time a song changes or some other state changes right?

```suggestion
    if info['track']:
      print(f"{info['state']}: {info['track']}")
    else:
      print(info['state'])
    prev_state = info['state']
```

Reply:
I did the check for a state change because it was ending up like this:
![image](https://user-images.githubusercontent.com/70664737/129099101-d79dc5c9-2583-4655-8ca9-cdf3ae8afeaa.png)

This was the easiest way I could think of to limit it to one message per action (each of those clusters of playing/pause are from one click of the button). Looking at it now, I'm going to have to check state and track together anyway because it won't show a new song transition if spotify continuously plays